### ISSUE TITLE
testing

### ISSUE THREAD

**Issue by wconnell** [wconnell/enformer-finetune#8]
hey @ssenan, when you have a moment it would be great to have you spin up a `ddp` run on your hardware. the readme has directions for getting setup.

thanks, let me know what you run into and I can help out.

**Comment by ssenan** [comment]
hey, I'll give it a try either tomorrow or Wednesday and let you know if I encounter any issues.

**Comment by ssenan** [comment]
Here is the current error I get when trying the main branch.
![enformer_error](https://github.com/wconnell/enformer-finetune/assets/8073222/e7ac14da-f3ae-4481-87bd-f574ae3139e1)

I have some clarifying questions regarding some of the sequence insertion strategy / code structure and wouldn't mind sitting it on your next meeting with Aaron to discuss if possible.

A side note sklearn is not installed via requirements, but is utilized during the data split process.

**Comment by wconnell** [comment]
file names of preprocessed data should be corrected now @ssenan

**Comment by ssenan** [comment]
Just putting this here for later reference:

Error with bfloat16 training:
<img width="854" alt="image" src="https://github.com/wconnell/enformer-finetune/assets/8073222/66b0e8ff-71d7-4842-b7ca-b494051dcb41">

Error with ddp_find_unused_paramters_true + precision = 16-mixed
<img width="855" alt="image" src="https://github.com/wconnell/enformer-finetune/assets/8073222/8914c77b-a4de-4900-83c3-9f3ec3563691">


I will spend some time exploring what is causing the need for the find_used_parameters, while simultaneously exploring some methods for decreasing the memory requirements for finetuning.

**Comment by aaronwtr** [comment]
[PR6](https://github.com/wconnell/enformer-finetune/pull/6) runs with batch size 4 for a couple of epochs. I am yet to complete a full (20 epochs) run because of long queue of the cluster scheduler I am using. 

The logs contain version_00 and version_0. Version_00 was trained with batch size 2 and version 0 with batch size 4. Both run successfully. Also, still encountering NaNs on epoch end for version_0, not for version_00. I did play around with the tensor types and precision between different runs (I believe the batch size 2 was able to run with 32 bit precision whereas batch size 4 needed to be scaled down to 16-mixed).

**Comment by wconnell** [comment]
nice – yeah this is reproducing the issue I had with nans in the loss at 16-bit precision!

in conclusion, I think we need to run at 32 bit. it would be great to get an 80 gb machine...I'm onboarding to Stability cluster but I'm not sure what resources are available yet

I will review the PR and close this issue with it. thanks @aaronwtr !
